sql/tests: TestRandomSyntaxSQLSmith failed [NPE in applyJoinNode]

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11947069?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11947069?buildTab=artifacts#/) on master @ [4b2badd7bc6c11a7646e66b131f5bcaf8328916a](https://github.com/cockroachdb/cockroach/commits/4b2badd7bc6c11a7646e66b131f5bcaf8328916a):

Random syntax error:

```
    rsg_test.go:895: Crash detected: server panic: driver: bad connection
```
Query:

```
        SELECT
        	tab378984.crdb_internal_mvcc_timestamp AS "%pcol857759",
        	'48 years 7 mons 894 days 13:39:26.674765':::INTERVAL AS col857760,
        	tab378983.tableoid AS "coL857761",
        	tab378983.crdb_internal_mvcc_timestamp AS col857762,
        	(SELECT (-4999644074744333745):::INT8 AS "co""l857763" LIMIT 1:::INT8) AS col857764,
        	tab378983._inet AS col857765,
        	'2011-06-28 16:37:44.000635+00':::TIMESTAMPTZ AS col857766,
        	tab378984._bool AS col857767,
        	tab378983.tableoid AS col857768,
        	tab378984._timestamptz AS col857769,
        	NULL AS "Co😽l857770",
        	tab378984._string AS "c)ol857771",
        	NULL AS col857772,
        	tab378983._int4 AS c🙃ol857773,
        	tab378984._interval AS " col857774",
        	tab378984.crdb_internal_mvcc_timestamp AS "c%69ol'857775"
        FROM
        	defaultdb.public.seed@seed__int8__float8__date_idx AS tab378983, defaultdb.public.seed@[0] AS tab378984
        WHERE
        	(true AND ('9E82DF40/BC8A8379':::PG_LSN::PG_LSN NOT IN (SELECT pg_lsn(tab378984._string::STRING)::PG_LSN::PG_LSN AS col857758 FROM defaultdb.public.seed@[0] AS "%ptAb%v378985" WHERE "%ptAb%v378985"._bool LIMIT 22:::INT8)))
        ORDER BY
        	tab378983._timestamptz ASC NULLS FIRST
        LIMIT
        	78:::INT8;
```
Schema:

```
    rsg_test.go:704: To reproduce, use schema:
    rsg_test.go:706: SET sql_safe_updates = false;;
    rsg_test.go:706: SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;;
    rsg_test.go:706: SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;;
    rsg_test.go:706: CREATE TABLE table1 (col1_0 INT4, col1_1 PG_LSN NOT NULL, """col1_2" CHAR NOT NULL, col1_3 FLOAT8 NOT NULL, PRIMARY KEY (col1_1 ASC, """col1_2", col1_3 ASC), UNIQUE (col1_3 DESC, (col1_0 + 1718159908:::INT8) ASC) WHERE table1.col1_3 = '+Inf':::FLOAT8, UNIQUE ((col1_0 + 1656506633:::INT8) DESC, col1_0) WHERE table1."""col1_2" <= '':::STRING, INDEX (lower("""col1_2") ASC) STORING (col1_0) NOT VISIBLE, INDEX (col1_0, col1_1 DESC, """col1_2", lower("""col1_2") ASC) VISIBILITY 0.80, UNIQUE (col1_3 ASC, col1_0 ASC, """col1_2") WHERE table1."""col1_2" != '"':::STRING, UNIQUE (col1_0 ASC, """col1_2" ASC, col1_3, col1_1 DESC) WHERE ((table1.col1_0 = 1:::INT8) OR (table1.col1_3 != (-1.0):::FLOAT8)) AND (table1."""col1_2" > e'\U00002603':::STRING));
    rsg_test.go:706: ALTER TABLE table1 INJECT STATISTICS e'[{"avg_size": 21, "columns": ["col1_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 93126895713, "histo_buckets": [{"distinct_range": 0, "num_eq": 80, "num_range": 0, "upper_bound": "-1760570338"}, {"distinct_range": 19584145439.95768, "num_eq": 8598521184409481157, "num_range": 40000000000, "upper_bound": "-1"}, {"distinct_range": 0, "num_eq": 6952282162260467974, "num_range": 2147040682226951462, "upper_bound": "320165031"}, {"distinct_range": 100000000000, "num_eq": 3728072327274856069, "num_range": 100000000000, "upper_bound": "338991470"}, {"distinct_range": 70000000000, "num_eq": 400000, "num_range": 70000000000, "upper_bound": "1266827709"}, {"distinct_range": 2376280076838793700, "num_eq": 7000, "num_range": 2545411094513238110, "upper_bound": "1605341739"}], "histo_col_type": "INT4", "name": "__auto__", "null_count": 98551908539, "row_count": 100000000000}, {"avg_size": 13, "columns": ["col1_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 8817961525, "histo_col_type": "PG_LSN", "name": "__auto__", "null_count": 0, "row_count": 100000000000}, {"avg_size": 7, "columns": ["\\"\\"\\"col1_2\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 60511047609, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 100000000000}, {"avg_size": 6, "columns": ["col1_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 7321223047, "histo_buckets": [{"distinct_range": 0, "num_eq": 8468293078890346225, "num_range": 0, "upper_bound": "-0.8111565926847524"}, {"distinct_range": 0, "num_eq": 6999228595962647056, "num_range": 1000, "upper_bound": "-0.2258070473470397"}, {"distinct_range": 0, "num_eq": 3000000000, "num_range": 30, "upper_bound": "-0.1557861596297135"}, {"distinct_range": 4040.509921801615, "num_eq": 8000, "num_range": 6000, "upper_bound": "1.401298464324817e-45"}], "histo_col_type": "FLOAT8", "name": "__auto__", "null_count": 0, "row_count": 100000000000}]':::JSONB;
    rsg_test.go:706: CREATE TYPE rand_typ_0 AS ENUM ('dtlbs');
    rsg_test.go:706: CREATE TYPE rand_typ_1 AS ENUM ('ttvhur', 'kjqsy', 'w', 'roehgu', 'ikwibk');
    rsg_test.go:706: SET sql_safe_updates = false;;
    rsg_test.go:706: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:706: ALTER TABLE seed SPLIT AT VALUES (5638265504913574179);
    rsg_test.go:706: ALTER TABLE seed SCATTER;
    rsg_test.go:708: 
    rsg_test.go:709: -- test log scope end --
--- FAIL: TestRandomSyntaxSQLSmith (370.88s)
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith2630168632
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31858

repro:
```
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
                SELECT
                        g::INT2 AS _int2,
                        g::INT4 AS _int4,
                        g::INT8 AS _int8,
                        g::FLOAT4 AS _float4,
                        g::FLOAT8 AS _float8,
                        '2001-01-01'::DATE + g AS _date,
                        '2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
                        '2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
                        g * '1 day'::INTERVAL AS _interval,
                        g % 2 = 1 AS _bool,
                        g::DECIMAL AS _decimal,
                        g::STRING AS _string,
                        g::STRING::BYTES AS _bytes,
                        substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
                        '0.0.0.0'::INET + g AS _inet,
                        g::STRING::JSONB AS _jsonb,
                        enum_range('hello'::greeting)[g] as _enum
                FROM
                        generate_series(1, 5) AS g;
        COMMIT;

        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        

SELECT tab378984.crdb_internal_mvcc_timestamp AS "%pcol857759", '48 years 7 mons 894 days 13:39:26.674765':::INTERVAL AS col857760, tab378983.tableoid AS "coL857761", tab378983.crdb_internal_mvcc_timestamp AS col857762, (SELECT (-4999644074744333745):::INT8 AS "co""l857763" LIMIT 1:::INT8) AS col857764, tab378983._inet AS col857765, '2011-06-28 16:37:44.000635+00':::TIMESTAMPTZ AS col857766, tab378984._bool AS col857767, tab378983.tableoid AS col857768, tab378984._timestamptz AS col857769, NULL AS "Co😽l857770", tab378984._string AS "c)ol857771", NULL AS col857772, tab378983._int4 AS c🙃ol857773, tab378984._interval AS " col857774", tab378984.crdb_internal_mvcc_timestamp AS "c%69ol'857775" FROM defaultdb.public.seed@seed__int8__float8__date_idx AS
tab378983, defaultdb.public.seed@[0] AS tab378984 WHERE (true AND ('9E82DF40/BC8A8379':::PG_LSN::PG_LSN NOT IN (SELECT pg_lsn(tab378984._string::STRING)::PG_LSN::PG_LSN AS col857758 FROM defaultdb.public.seed@[0] AS "%ptAb%v378985" WHERE "%ptAb%v378985"._bool LIMIT 22:::INT8))) ORDER BY tab378983._timestamptz ASC NULLS FIRST LIMIT 78:::INT8;
```

stack trace is:
```
* ERROR: a panic has occurred!
* runtime error: invalid memory address or nil pointer dereference
* (1) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:952
*   | [...repeated from below...]
* Wraps: (2) while executing: SELECT _._ AS _, '_':::INTERVAL AS _, _._ AS _, _._ AS _, (SELECT (_):::INT8 AS _ LIMIT _:::INT8) AS _, _._ AS _, '_':::TIMESTAMPTZ AS _, _._ AS _, _._ AS _, _._ AS _, _ AS _, _._ AS _, _ AS _, _._ AS _, _._ AS _, _._ AS _ FROM _@_ AS _, _@[0] AS _ WHERE (_ AND ('_':::PG_LSN::PG_LSN NOT IN (SELECT pg_lsn(_._::STRING)::PG_LSN::PG_LSN AS _ FROM _@[0] AS _ WHERE _._ LIMIT _:::INT8))) ORDER BY _._ ASC NULLS FIRST LIMIT _:::INT8
* Wraps: (3) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:952
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:884
*   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:884
*   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:884
*   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:884
*   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:884
*   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:884
*   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:884
*   | runtime.panicmem
*   | 	GOROOT/src/runtime/panic.go:260
*   | runtime.sigpanic
*   | 	GOROOT/src/runtime/signal_unix.go:841
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLReceiver).resetForLocalRerun
*   | 	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1245
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun
*   | 	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:2005
*   | github.com/cockroachdb/cockroach/pkg/sql.runPlanInsidePlan
*   | 	github.com/cockroachdb/cockroach/pkg/sql/apply_join.go:353
*   | github.com/cockroachdb/cockroach/pkg/sql.(*applyJoinNode).runNextRightSideIteration
*   | 	github.com/cockroachdb/cockroach/pkg/sql/apply_join.go:258
*   | github.com/cockroachdb/cockroach/pkg/sql.(*applyJoinNode).Next
*   | 	github.com/cockroachdb/cockroach/pkg/sql/apply_join.go:223
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Next
*   | 	github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:220
*   | github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Columnarizer).Next
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colexec/columnarizer.go:239
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).next
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/stats.go:118
*   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).Next
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/stats.go:126
*   | github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecjoin.(*crossJoiner).consumeRightInput
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecjoin/crossjoiner.go:183
*   | github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecjoin.(*crossJoiner).Next
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecjoin/crossjoiner.go:99
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).next
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/stats.go:118
*   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).Next
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/stats.go:126
*   | github.com/cockroachdb/cockroach/pkg/sql/colexec.spool_false
*   | 	github.com/cockroachdb/cockroach/bazel-out/darwin_arm64-fastbuild/bin/pkg/sql/colexec/sorttopk.eg.go:321
* Wraps: (4) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) *safedetails.withSafeDetails (3) *withstack.withStack (4) runtime.errorString
*
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10556ec7c]

goroutine 2497 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0x14005d6cc80, {0x109374e60, 0x14005785f80}, {0x1089d8fc0?, 0x10c5dd8a0?})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1199 +0x164
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1({0x109374e60, 0x14005785f80}, {0xb?})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:952 +0x4c
panic({0x1089d8fc0, 0x10c5dd8a0})
	GOROOT/src/runtime/panic.go:884 +0x204
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58 +0x340
panic({0x1089d8fc0, 0x10c5dd8a0})
	GOROOT/src/runtime/panic.go:884 +0x204
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58 +0x340
panic({0x1089d8fc0, 0x10c5dd8a0})
	GOROOT/src/runtime/panic.go:884 +0x204
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58 +0x340
panic({0x1089d8fc0, 0x10c5dd8a0})
	GOROOT/src/runtime/panic.go:884 +0x204
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58 +0x340
panic({0x1089d8fc0, 0x10c5dd8a0})
	GOROOT/src/runtime/panic.go:884 +0x204
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58 +0x340
panic({0x1089d8fc0, 0x10c5dd8a0})
	GOROOT/src/runtime/panic.go:884 +0x204
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58 +0x340
panic({0x1089d8fc0, 0x10c5dd8a0})
	GOROOT/src/runtime/panic.go:884 +0x204
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLReceiver).resetForLocalRerun(0x14004a6d880, {0x14005094d20?, 0x14005094d00?, 0x1?, 0x14005440460?})
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1245 +0xbc
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0x14000df6780, {0x109374e60, 0x14005094d20}, 0x140045cd7b8, 0x14005440fa0, 0x14005440f00?, {{0x109378238, 0x14005440f00}, 0x0}, 0x14004a6d880, ...)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:2005 +0x328
github.com/cockroachdb/cockroach/pkg/sql.runPlanInsidePlan({0x109374e60, 0x14005094d20}, {{0x109374e60, 0x14005094540}, 0x14003e4d800, 0x14005d6d2d8}, 0x140012e4310, {0x1093769b8, 0x14003aed0e0}, {0x0?, ...})
	github.com/cockroachdb/cockroach/pkg/sql/apply_join.go:353 +0x5d0
github.com/cockroachdb/cockroach/pkg/sql.(*applyJoinNode).runNextRightSideIteration(0x1400609c000, {{0x109374e60?, 0x14005094540?}, 0x14003e4d800?, 0x14005d6d2d8?}, {0x14005294dc0, 0x5, 0x5})
	github.com/cockroachdb/cockroach/pkg/sql/apply_join.go:258 +0x1ec
github.com/cockroachdb/cockroach/pkg/sql.(*applyJoinNode).Next(0x1400609c000, {{0x109374e60, 0x14005094540}, 0x14003e4d800, 0x14005d6d2d8})
	github.com/cockroachdb/cockroach/pkg/sql/apply_join.go:223 +0x164
github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Next(0x14004016000)
	github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:220 +0x144
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Columnarizer).Next(0x14002188600)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/columnarizer.go:239 +0xf4
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).next(...)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/stats.go:118
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0x14004527080?)
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92 +0x58
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).Next(0x14007617700)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/stats.go:126 +0x48
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecjoin.(*crossJoiner).consumeRightInput(0x140049bca80, {0x109374e60, 0x14005094420})
	github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecjoin/crossjoiner.go:183 +0x1d8
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecjoin.(*crossJoiner).Next(0x140049bca80)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecjoin/crossjoiner.go:99 +0x54
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).next(...)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/stats.go:118
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0x14004527200?)
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92 +0x58
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).Next(0x14007617800)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/stats.go:126 +0x48
github.com/cockroachdb/cockroach/pkg/sql/colexec.spool_false(0x14004a729c0)
	github.com/cockroachdb/cockroach/bazel-out/darwin_arm64-fastbuild/bin/pkg/sql/colexec/sorttopk.eg.go:321 +0x34
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*topKSorter).spool(0x1400624c198?)
	github.com/cockroachdb/cockroach/bazel-out/darwin_arm64-fastbuild/bin/pkg/sql/colexec/sorttopk.eg.go:73 +0x2c
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*topKSorter).Next(0x14004a729c0)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/sorttopk.go:165 +0x5c
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecdisk.(*diskSpillerBase).Next.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecdisk/disk_spiller.go:202 +0x34
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0x135d81958?)
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92 +0x58
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecdisk.(*diskSpillerBase).Next(0x1400284bc20)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecdisk/disk_spiller.go:200 +0x54
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecutils.(*vectorTypeEnforcer).Next(0x14005294f00)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecutils/operator.go:157 +0x30
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase.constIntervalOp.Next({{{{0x10938ff48, 0x14005294f00}}, {{0x109374e60, 0x14005094420}}}, 0x14004527400, 0xa, {0x247, 0x37e, 0x2cb7822f24c8}})
	github.com/cockroachdb/cockroach/bazel-out/darwin_arm64-fastbuild/bin/pkg/sql/colexec/colexecbase/const.eg.go:485 +0x50
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecutils.(*vectorTypeEnforcer).Next(0x14005294fa0)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecutils/operator.go:157 +0x30
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase.constInt64Op.Next({{{{0x10938ff48, 0x14005294fa0}}, {{0x109374e60, 0x14005094420}}}, 0x14004527440, 0xb, 0xba9db234097a3a4f})
	github.com/cockroachdb/cockroach/bazel-out/darwin_arm64-fastbuild/bin/pkg/sql/colexec/colexecbase/const.eg.go:368 +0x48
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecutils.(*vectorTypeEnforcer).Next(0x14005294ff0)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecutils/operator.go:157 +0x30
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase.constTimestampOp.Next({{{{0x10938ff48, 0x14005294ff0}}, {{0x109374e60, 0x14005094420}}}, 0x140045274c0, 0xc, {0x9b078, 0xec59bfa58, 0x14000bca070}})
	github.com/cockroachdb/cockroach/bazel-out/darwin_arm64-fastbuild/bin/pkg/sql/colexec/colexecbase/const.eg.go:446 +0x50
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecutils.(*vectorTypeEnforcer).Next(0x14005295090)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecutils/operator.go:157 +0x30
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase.constNullOp.Next({{{{0x10938ff48, 0x14005295090}}, {{0x109374e60, 0x14005094420}}}, 0xd})
	github.com/cockroachdb/cockroach/bazel-out/darwin_arm64-fastbuild/bin/pkg/sql/colexec/colexecbase/const.eg.go:611 +0x3c
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase.(*simpleProjectOp).Next(0x140047f6600)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase/simple_project.go:124 +0x34
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).next(...)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/stats.go:118
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0x14004527580?)
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92 +0x58
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).Next(0x14007617900)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/stats.go:126 +0x48
github.com/cockroachdb/cockroach/pkg/sql/colexecop.(*noopOperator).Next(0x102879d10?)
	github.com/cockroachdb/cockroach/pkg/sql/colexecop/operator.go:430 +0x28
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase.(*simpleProjectOp).Next(0x140047f6660)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase/simple_project.go:124 +0x34
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).next(...)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/stats.go:118
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0x14004527680?)
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92 +0x58
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).Next(0x14007617b00)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/stats.go:126 +0x48
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).nextAdapter(...)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:250
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0x109374db8?)
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92 +0x58
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).next(...)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:254
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).Run(0x14004f58640, {0x109374db8?, 0x14005294d70?})
	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:286 +0x10c
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run(0x14005b5a240, {0x109374db8, 0x14005294d70}, 0x0)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:320 +0x1a8
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0x14000df6780, {0x109374e60, 0x140011251d0}, 0x14005440aa0, 0x14005440460, 0x140049bc680, 0x14004480000, 0x14005d6d790, 0x14004229c50)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:906 +0x8a0
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0x14000df6780, {0x109374e60, 0x140011251d0}, 0x14005d6d790, 0x14005440aa0, 0x1055611b8?, {{0x109378238, 0x14005440a00}, 0x0}, 0x14004480000, ...)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1957 +0x170
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll.func3(0x14005d6d2d8, 0x0?, {0x109374e60, 0x140011251d0}, 0x1400624d958?, 0x102816398?, 0x0?)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1672 +0x84
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll(0x14005440aa0?, {0x109374e60?, 0x140011251d0}, 0x14005d6d790, 0x14005440aa0, 0x14005d6d2d8, 0x14004480000, 0x0)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1675 +0x17c
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0x14005d6cc80, {0x109374e60?, 0x140011251d0}, 0x14005d6d2d8, 0x0?, {0x135d8b010?, 0x14000acfb28}, 0x1400624de50?, 0x140062942d8)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:2291 +0x480
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0x14005d6cc80, {0x109374e60, 0x14004f147e0}, 0x14005d6d2d8, {0x135d8b010, 0x14000acfb28})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1839 +0xe44
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0x14005d6cc80, {0x109374e60, 0x14004f147e0}, {{0x1093a1940, 0x14005a2f090}, {0x0, 0x0, 0x0}, {0x140030b7475, 0x450}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1070 +0x33d4
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt.func1({0x109374e60?, 0x14004f14660?})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:144 +0x8c
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithProfiling(0x109374e60?, {0x109374e60?, 0x14004f14660?}, {0x1093a1940?, 0x14005a2f090?}, 0x0?, 0x0?)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:3235 +0x280
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0x14005d6cc80, {0x109374e60, 0x14004f14660}, {{0x1093a1940, 0x14005a2f090}, {0x0, 0x0, 0x0}, {0x140030b7475, 0x450}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:143 +0x2e8
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1(0x1400624f970, 0x1400624f598, 0x14005d6cc80, 0x1400624f4c8, 0x14006c6d5b8, 0x14006c6d5a8, 0x14006c6d5c8, 0x1400624f5d8)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2220 +0x2d4
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0x14005d6cc80)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2225 +0x28c
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0x14005d6cc80, {0x109374e60, 0x14005785f80}, 0x1032181b8?, 0x14003b20410?, 0x14000d48c80?)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:2142 +0x170
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0x140020a3f40?, {0x109374e60?, 0x14005785f80?}, {0x1400256102a?}, 0x4?, 0x14005785e00?)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:954 +0xb4
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommands(0x14000acf400, {0x109374e60, 0x14005785f80}, {0x1, 0x2, {0x1, {0x106de8e73, 0x3}, {0x14003bf63d0, 0xf}, ...}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:247 +0x2a4
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).serveImpl.func3()
	github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:999 +0xcc
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).serveImpl
	github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:996 +0x498
```

--- a/pkg/sql/distsql_running.go
+++ b/pkg/sql/distsql_running.go
@@ -1243,7 +1243,9 @@ func (r *DistSQLReceiver) resetForLocalRerun(stats topLevelQueryStats) {
 	r.closed = false
 	r.stats = stats
 	r.egressCounter = nil
-	atomic.StoreUint64(r.progressAtomic, math.Float64bits(0))
+	if r.progressAtomic != nil {
+		atomic.StoreUint64(r.progressAtomic, math.Float64bits(0))
+	}
 }
 
 // Release releases this DistSQLReceiver back to the pool.

--- a/pkg/sql/logictest/testdata/logic_test/apply_join
+++ b/pkg/sql/logictest/testdata/logic_test/apply_join
@@ -665,3 +665,74 @@ CREATE MATERIALIZED VIEW v1 AS
             UNION
             SELECT aid, pid FROM cte2
         );
+
+# Regression test for #111327 - the query shouldn't cause a nil-pointer error.
+statement ok
+CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning');
+
+statement ok
+CREATE TABLE IF NOT EXISTS seed AS
+SELECT
+  g :: INT2 AS _int2,
+  g :: INT4 AS _int4,
+  g :: INT8 AS _int8,
+  g :: FLOAT4 AS _float4,
+  g :: FLOAT8 AS _float8,
+  '2001-01-01' :: DATE + g AS _date,
+  '2001-01-01' :: TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
+  '2001-01-01' :: TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
+  g * '1 day' :: INTERVAL AS _interval,
+  g % 2 = 1 AS _bool,
+  g :: DECIMAL AS _decimal,
+  g :: STRING AS _string,
+  g :: STRING :: BYTES AS _bytes,
+  substring('00000000-0000-0000-0000-' || g :: STRING || '00000000000', 1, 36):: UUID AS _uuid,
+  '0.0.0.0' :: INET + g AS _inet,
+  g :: STRING :: JSONB AS _jsonb,
+  enum_range('hello' :: greeting) [g] as _enum
+FROM
+  generate_series(1, 5) AS g;
+
+statement ok
+INSERT INTO seed DEFAULT VALUES;
+
+statement ok
+CREATE INDEX on seed (_int8, _float8, _date);
+
+statement ok
+CREATE INVERTED INDEX on seed (_jsonb);
+
+skipif config local-mixed-23.1
+statement error pgcode 22P02 pq: pg_lsn\(\): invalid input syntax for type pg_lsn: \"1\"
+SELECT
+  tab378984.crdb_internal_mvcc_timestamp AS "%pcol857759",
+  '48 years 7 mons 894 days 13:39:26.674765':::INTERVAL AS col857760,
+  tab378983.tableoid AS "coL857761",
+  tab378983.crdb_internal_mvcc_timestamp AS col857762,
+  (SELECT (-4999644074744333745):::INT8 AS "co""l857763" LIMIT 1:::INT8) AS col857764,
+  tab378983._inet AS col857765,
+  '2011-06-28 16:37:44.000635+00':::TIMESTAMPTZ AS col857766,
+  tab378984._bool AS col857767,
+  tab378983.tableoid AS col857768,
+  tab378984._timestamptz AS col857769,
+  NULL AS "Co😽l857770",
+  tab378984._string AS "c)ol857771",
+  NULL AS col857772,
+  tab378983._int4 AS c🙃ol857773,
+  tab378984._interval AS " col857774",
+  tab378984.crdb_internal_mvcc_timestamp AS "c%69ol'857775"
+FROM
+  seed@seed__int8__float8__date_idx AS tab378983,
+  seed@[0] AS tab378984
+WHERE
+  (
+    true
+    AND (
+      '9E82DF40/BC8A8379':::PG_LSN::PG_LSN NOT IN (
+        SELECT pg_lsn(tab378984._string::STRING)::PG_LSN::PG_LSN AS col857758
+        FROM seed@[0] AS "%ptAb%v378985" WHERE "%ptAb%v378985"._bool LIMIT 22:::INT8
+      )
+    )
+  )
+ORDER BY tab378983._timestamptz ASC NULLS FIRST
+LIMIT 78:::INT8;

--- a/pkg/sql/logictest/testdata/logic_test/pg_lsn
+++ b/pkg/sql/logictest/testdata/logic_test/pg_lsn
@@ -4,12 +4,6 @@ SELECT 'A01F0/1AAA'::pg_lsn
 ----
 A01F0/1AAA
 
-statement error could not parse pg_lsn
-SELECT 'A/G'::pg_lsn
-
-statement error could not parse pg_lsn
-SELECT '0G'::pg_lsn
-
 statement ok
 CREATE TABLE pg_lsn_table(id pg_lsn PRIMARY KEY, val pg_lsn)
 
@@ -80,3 +74,22 @@ SELECT * FROM ( VALUES
 23021024706816
 18446744073709551615
 -18446744073709551615
+
+statement error pgcode 22P02 pq: invalid input syntax for type pg_lsn: \"A/G\"
+SELECT 'A/G'::pg_lsn
+
+statement error pgcode 22P02 pq: invalid input syntax for type pg_lsn: \"0G\"
+SELECT '0G'::pg_lsn
+
+statement error pgcode 22P02 pq: invalid input syntax for type pg_lsn: \"ab\"
+SELECT 'ab'::PG_LSN;
+
+statement error pgcode 22P02 pq: pg_lsn\(\): invalid input syntax for type pg_lsn: \"ab\"
+SELECT pg_lsn('ab');
+
+statement error pgcode 22P02 pq: invalid input syntax for type pg_lsn: \"1\"
+SELECT '1'::PG_LSN;
+
+statement error pgcode 22P02 pq: invalid input syntax for type pg_lsn: \"\"
+SELECT ''::PG_LSN;
+

--- a/pkg/sql/sem/tree/constant_test.go
+++ b/pkg/sql/sem/tree/constant_test.go
@@ -221,7 +221,8 @@ func TestStringConstantVerifyAvailableTypes(t *testing.T) {
 
 			semaCtx := tree.MakeSemaContext()
 			if _, err := test.c.ResolveAsType(context.Background(), &semaCtx, availType); err != nil {
-				if !strings.Contains(err.Error(), "could not parse") {
+				if !strings.Contains(err.Error(), "could not parse") &&
+					!strings.Contains(err.Error(), "invalid input syntax") {
 					// Parsing errors are permitted for this test, as proper tree.StrVal parsing
 					// is tested in TestStringConstantTypeResolution. Any other error should
 					// throw a failure.
@@ -693,7 +694,8 @@ func TestStringConstantResolveAvailableTypes(t *testing.T) {
 					if !strings.Contains(err.Error(), "could not parse") &&
 						!strings.Contains(err.Error(), "parsing") &&
 						!strings.Contains(err.Error(), "out of range") &&
-						!strings.Contains(err.Error(), "exceeds supported") {
+						!strings.Contains(err.Error(), "exceeds supported") &&
+						!strings.Contains(err.Error(), "invalid input syntax") {
 						// Parsing errors are permitted for this test, but the number of correctly
 						// parseable types will be verified. Any other error should throw a failure.
 						t.Errorf("%d: expected resolving %v as available type %s would either succeed"+

--- a/pkg/sql/sem/tree/datum.go
+++ b/pkg/sql/sem/tree/datum.go
@@ -3629,7 +3629,9 @@ func NewDPGLSN(l lsn.LSN) *DPGLSN {
 func ParseDPGLSN(str string) (*DPGLSN, error) {
 	v, err := lsn.ParseLSN(str)
 	if err != nil {
-		return nil, errors.Wrapf(err, "could not parse pg_lsn")
+		return nil, pgerror.Newf(pgcode.InvalidTextRepresentation,
+			"invalid input syntax for type pg_lsn: \"%s\"", str,
+		)
 	}
 	return NewDPGLSN(v), nil
 }

